src/fread.c

```diff
@@ -2035,7 +2035,10 @@ int freadMain(freadMainArgs _args) {
     }
     if (verbose) {
       if (sampleLines==0) {
-        DTPRINT(_("  'header' determined to be %s because there are%s number fields in the first and only row\n"), args.header?"TRUE":"FALSE", args.header?_(" no"):"");
+        if (args.header)
+          DTPRINT(_("  'header' determined to be true because there are no number fields in the first and only row\n"));
+        else
+          DTPRINT(_("  'header' determined to be false because there are number fields in the first and only row\n"));
```

Comment:
```suggestion
          DTPRINT(_("  'header' determined to be TRUE because there are no number fields in the first and only row\n"));
        else
          DTPRINT(_("  'header' determined to be FALSE because there are number fields in the first and only row\n"));
```